_posts/2019-08-23-unet/unet.Rmd

```diff
@@ -0,0 +1,437 @@
+ 
+author:
+  - name: Daniel Falbel and Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: falbelkeydana2019unet
+date: 08-23-2019
+categories:
+  - Images
+  - Introductions
+  - Keras
+bibliography: bibliography.bib
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/unet.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+Sure, it is nice when I have a picture of some object, and a neural network can tell me what kind of object that is. More realistically, there might be several salient objects in that picture, and it tells me what they are, and where they are. The latter task especially (known as _object detection_) seems prototypical of contemporary AI applications that at the same time are fascinating intellectually and questionable ethically. It's different with the subject of this post: Successful _image segmentation_ has a lot of undeniably useful applications. For example, it is a sine qua non in medicine, neuroscience, biology and other life sciences.
+
+So what, technically, is image segmentation, and how can we train a neural network to do it?
+
+## Image segmentation in a nutshell
+
+Say we have an image with a bunch of cats in it. In _classification_, the question is "what's that?", and the answer we want to hear is: "cat". In _object detection_, we again ask "what's that", but now that "what" is implicitly plural, and we expect an answer like "there's a cat, a cat and a cat, and they're here, here, and here" (imagine the network pointing, by means of drawing _bounding boxes_, i.e., rectangles around the detected objects). In _segmentation_, we want more: We want the whole image covered by "boxes" -- which aren't boxes anymore, but unions of pixel-size "boxlets" -- or put differently: __We want the network to label every single pixel in the image.__
+
+Here's an example from the paper we're going to talk about in a second. On the left is the input image (HeLa cells), next up is the ground truth, and third is the learned segmentation mask.
+
+```{r, eval=TRUE, echo=FALSE, layout="l-body-outset", fig.cap = "Example segmentation from Ronneberger et al. 2015."}
+knitr::include_graphics("images/mask.png")
+```
+
+Technically, a distinction is made between _class segmentation_ and _instance segmentation_. In class segmentation, referring to the "bunch of cats" example, there are two possible labels: Every pixel is either "cat" or "not cat". Instance segmentation is more difficult: Here every cat gets their own label. (As an aside, why should that be more difficult? Presupposing human-like cognition, it wouldn't be -- if I have the concept of a cat, instead of just "cattiness", I "see" there are two cats, not one. But depending on what a specific neural network relies on most -- texture, color, isolated parts -- those tasks may differ a lot in difficulty.)
```

Comment:
I really like this aside, and (if it doesn't come up again later) you might consider making it _not_ an aside — give it a bit of room of its own.

Reply:
Thanks! Honestly I think that in this post, it's better to leave it  as an "aside".
But perhaps in the future, some time, you know, there could be posts directly aimed at discussing such things :-)